pkg/api/v20210131preview/openshiftcluster_validatestatic.go

```diff
@@ -210,6 +211,14 @@ func (sv *openShiftClusterStaticValidator) validateNetworkProfile(path string, n
 		}
 	}
 
+	switch np.SDNProvider {
+	case SDNProviderOVNKubernetes, SDNProviderOpenShiftSDN:
+	default:
+		fmt.Println(np.SDNProvider)
+		errorMsg := fmt.Sprintf("The provided SDNProvider must be either '%s' or '%s'.", SDNProviderOVNKubernetes, SDNProviderOpenShiftSDN)
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".sdnProvider", errorMsg)
```

Comment:
If you look inside `NewCloudError` it is already doign message merging with `message` and `args...` so `Sprintf` here is redundant. 
```
func NewCloudError(statusCode int, code, target, message string, a ...interface{}) *CloudError {
	return &CloudError{
		StatusCode: statusCode,
		CloudErrorBody: &CloudErrorBody{
			Code:    code,
			Message: fmt.Sprintf(message, a...),
			Target:  target,
		},
	}
}
```

Reply:
Sorry, I didn't see that. Thanks for pointing it out!